grammar/parsing.js

```diff
@@ -206,12 +206,80 @@ class Parsing extends EmbeddedActionsParser {
             return result;
         });
 
+        $.RULE('multiArray', () => {
+            const multiArr = [[]];
+            let singleArr = null;
+
+            const updateRow = (multiArr, newArr) => {
+                if (multiArr[0] === null)
```

Comment:
Not sure if this check is necessary given the initial value assigned to `multiArr` on L210. However, if this function is to be reused in other situations, it's good to have some sort of sanity check.

The bigger problem is I'm not sure it's going to work as intended.  L214 and L215 will do exactly one thing:
if `multiArr = [null]` (could have more than one element; the first one just has to be exactly `null`), it will update it to be `[null, []]` (again, could have more than two elements, but the last one is gonna be an empty array).

Note that it the condition `multiArr[0] === null` will _not_ be met by `multiArr = []` because of the `===`. If you just want to check if the first value is empty, you could potentially do `multiArr[0] == null`, as `== null` matches exactly two things `null` and `undefined`.

But, ultimately, why not what is used below?: 

```js
if (!Array.isArray(multiArr[0]))
```

Note: if we are concerned that `multiArr` might not be an array at all (in which case, checking `multiArr[0]` will throw an undefined error), we'd actually need an additional check for that: 

```
if (!Array.isArray(multiArr)) { 
  // bad input. throw error since we assume multiArr is a const, so it can't be redefined as an array 
}
```